proxy-node-shared/src/main/java/uk/gov/ida/notification/validations/ValidSamlId.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.ida.notification.validations;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.Size;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Documented
+@Constraint(validatedBy = { SamlIdValidator.class })
+@Size(min = SamlIdValidator.MinLength, max = SamlIdValidator.MaxLength)
+@Target({ FIELD, PARAMETER })
+@Retention(RUNTIME)
+public @interface ValidSamlId {
```

Comment:
I like that you've created annotations for the validation a lot.  Is it possible to be more specific with the message where appropriate so we can see instantly exactly what caused the validation to fail?

Reply:
I'll do a little digging in case it's possible, but it didn't seem trivial to adjust the message from the annotation. The solution there might be to compose a validation from several smaller ones. It's quite boilerplate-y, though, which is why I'm hesitating...

Reply:
It's possible 👍